cognite/client/data_classes/workflows.py

```diff
@@ -1381,34 +1381,47 @@ class WorkflowTriggerRun(CogniteResource):
 
     def __init__(
         self,
-        trigger_external_id: str,
-        trigger_fire_time: int,
+        external_id: str,
+        fire_time: int,
         workflow_external_id: str,
         workflow_version: str,
+        workflow_execution_id: str,
+        status: Literal["success", "failed"],
+        reason_for_failure: str | None = None,
```

Comment:
Ideally also a literal.

Reply:
There can be many reasons, and new reasons added at any time in the server. Why a literal?

Reply:
Just help for the user. Literals are just hints/docs, so will not break if you suddenly send something else. You know the intentions the best and it might make sense to just keep it as a string.

Reply:
The reason is a sentence, it can come from other APIs, which have no control of (transformations, functions, simulators, ...)
So I think we keep it as a string.